stellargraph/data/explorer.py

```diff
@@ -743,7 +758,10 @@ def run(self, nodes, n_size, n=1, seed=None):
                             if len(neigh_et) > 0:
                                 samples = rs.choices(neigh_et, k=n_size[depth - 1])
                             else:  # this doesn't happen anymore, see the comment above
-                                samples = [None] * n_size[depth - 1]
+                                _size = n_size[depth - 1]
+                                samples = (
```

Comment:
Maybe this could be factored slightly differently to reduce the black explosion, and also maybe be a bit clearer due to naming the `-1`/`None` value:

```python
missing_sentinel = -1 if self.use_ilocs else None
samples = [missing_sentinel] * n_size[depth - 1]
```

Also, there's a comment "# this doesn't happen anymore, see the comment above", so is this actually doing anything?

(If it's not, maybe updating/removing the comment could be a separate PR, because it looks like it may require updating a few places.)

Reply:
I just changed all the sample BFS to always use ilocs so there's no black explosion anymore :)

Yeah as far as I can tell that comment is wrong and "this" is still happening.